FoxHound.Client/src/Posts/PostContentEditor.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { useState } from "react";
+import MarkdownEditor from "react-simplemde-editor";
+import hljs from "highlight.js";
+
+import "easymde/dist/easymde.min.css";
+import "highlight.js/styles/solarized-light.css";
+
+interface IProps {
+  id: string;
+  content: string;
+  setContent: React.Dispatch<React.SetStateAction<string>>;
```

Comment:
I spent a little time looking this up and I think you're right here. I was trying to find a definitive reason why. This is the best i found from the [docs](https://reactjs.org/docs/react-component.html#setstate)

> Think of setState() as a request rather than an immediate command to update the component. For better perceived performance, React may delay it, and then update several components in a single pass. React does not guarantee that the state changes are applied immediately.

So i think that makes sense to keep all setState() in the component that owns it.